fili-core/src/main/java/com/yahoo/bard/webservice/application/AbstractBinderFactory.java

```diff
@@ -387,6 +399,33 @@ protected void configure() {
         };
     }
 
+    /**
+     * Bind the components necessary to support protocol metrics.
+     *
+     * @param binder The binder to bind the generator to.
+     */
+    private void bindMetricGenerator(AbstractBinder binder) {
+        List<String> protocols = Collections.emptyList();
+        TypeLiteral<List<String>> stringListLiteral = new TypeLiteral<List<String>>() { };
+        binder.bind(protocols).named(NAME_ACTIVE_PROTOCOLS).to(stringListLiteral);
+
+        binder.bind(ApiMetricAnnotater.NO_OP_ANNOTATER).to(ApiMetricAnnotater.class);
```

Comment:
If we are simply using the ApiRequestLogicalMetricBinder from BardConfigResources, do we need to provide named bindings for the classes? I think just a hook in the ABF would suffice.

Reply:
I thought these might be useful to people who want to build their own component.  Think of them as documentary bindings.